scripts/registry-validate-source.py

```
@@ -76,22 +83,60 @@ def _validate_integrations(doc: dict) -> None:
+        if not isinstance(item, dict):
+            raise ValidationError(f"manifest.shell_init[{idx}] must be a table")
+        prefix = f"manifest.shell_init[{idx}]"
+        name = _expect_non_empty_str(item, "name", prefix)
+        binary = _expect_non_empty_str(item, "binary", prefix)
+        strategy = _expect_non_empty_str(item, "strategy", prefix)
+        if strategy not in SHELL_INIT_STRATEGY_VALUES:
+            raise ValidationError(f"{prefix}.strategy must be a supported shell init strategy")
+        if binary not in binaries:
+            raise ValidationError(f"{prefix}.binary must match an artifact binary")
+        if not any(field in item for field in SHELL_INIT_FIELDS):
+            raise ValidationError(f"{prefix} must declare at least one shell args field")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)